### ISSUE TITLE
Energy dashboard: energy cost entity not saveable

### ISSUE THREAD

**Issue by JazekerXX** [home-assistant/frontend#17369]
### Checklist

- [X] I have updated to the latest available Home Assistant version.
- [X] I have cleared the cache of my browser.
- [X] I have tried a different browser to see if it is related to my browser.

### Describe the issue you are experiencing

Hi, when i try to add an entity that tracks the cost of electricity, the save button does not lit up. the weird thing is that for gas cost i did the same and that works without a problem.!

here is how i set up the helper entity. it’s a float and the unit of measurement is €/kWh. should be ok right?..
![image](https://github.com/home-assistant/frontend/assets/41451684/f116b6e6-c630-42a4-a31b-35b61b67f5f5)

### Describe the behavior you expected

the save button should lit up, so i can save it..

### Steps to reproduce the issue

1. go to settings -> dashboards -> energy dashboard
2. edit an electricity usage or return entity
3. select option entity that tracks the cost
...

### What version of Home Assistant Core has the issue?

2023.7.2

### In which browser are you experiencing the issue with?

chrome, edge

### Which operating system are you using to run this browser?

win11

### State of relevant entities

```txt
entity: input_number.elektra_prijs
state: 0.39445

attributes:
initial: null
editable: true
min: 0
max: 1
step: 0.00001
mode: box
unit_of_measurement: €/kWh
icon: mdi:lightning-bolt
friendly_name: Elektra prijs
```

### Problem-relevant frontend configuration

```yaml
this is configured in the ui.
```

### Javascript errors shown in your browser console/inspector

```txt
no errors
```

### Additional information

![image](https://github.com/home-assistant/frontend/assets/41451684/cf0d4885-b6c7-47a5-b603-8165bdcd4649)
save button not lit.

**Comment by karwosts** [comment]
This is a bug that the statistic id gets cleared every time you re-open the editor. You can see the first dropdown box in your screenshot is  empty. 

If you re-select the statistic it should work. 

I believe this is already fixed pending 2023.8

**Comment by JazekerXX** [comment]
YES! that was the problem. thanks!
totally looked over, and did not expect that..

i guess i can close this one, since it's allready a known issue..
